music_assistant/providers/opensubsonic/sonic_provider.py

```diff
@@ -330,9 +333,12 @@ async def search(
     async def get_library_artists(self) -> AsyncGenerator[Artist, None]:
         """Provide a generator for reading all artists."""
         artists = await self._run_async(self._conn.get_artists)
-        for index in artists.index:
-            for artist in index.artist:
-                yield parse_artist(self.instance_id, artist)
+
+        if artists.index:
+            for index in artists.index:
+                if index.artist:
```

Comment:
```suggestion
                if not index.artist:
                    continue
```

a guard clause to prevent the nested indentation